### PR TITLE
Add details to secure Debian platform

### PR DESCRIPTION
**Pull request by smau-centreon**
## Description

Add details to secure Debian platform

## Target version

- [ ] 20.10.x (staging)
- [ ] 21.04.x (staging)
- [ ] 21.10.x (staging)
- [x] 22.04.x (staging)
- [ ] Cloud (staging)
- [ ] Plugin Packs (staging)
- [ ] 22.10.x (next)

### CODE REVIEW

File: versioned_docs/version-22.04/installation/prerequisites.md [hunk 1]
```diff
@@ -182,6 +182,20 @@ Description of software and linked users:
 | Centreon Engine   | centengine       | centreon-engine  | automatic start |
 | Centreon Gorgone  | gorgoned         | centreon-gorgone | automatic start |
 </TabItem>
+
+<TabItem value="Debian 11" label="Debian 11">
+
+| Software          | Service          | User             | Comment         |
+| ----------------- | ---------------- | ---------------- | --------------- |
+| Apache            | httpd            |www-data          | automatic start |
```
Comment (cg-tw) [inline versioned_docs/version-22.04/installation/prerequisites.md:190]:
```suggestion
| Apache            | httpd            | www-data          | automatic start |
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 2]
```diff
@@ -932,6 +1088,45 @@ yum install httpd24-nghttp2
...
+
+```diff
+-LoadModule mpm_prefork_module modules/mod_mpm_prefork.so
++#LoadModule mpm_prefork_module modules/mod_mpm_prefork.so
+
+-#LoadModule mpm_event_module modules/mod_mpm_event.so
++LoadModule mpm_event_module modules/mod_mpm_event.so
+```
+
+5. Restart the Apache process to take in account the new configuration:
```
Comment (cg-tw) [inline versioned_docs/version-22.04/administration/secure-platform.md:1124]:
```suggestion
5. Restart the Apache process to take into account the new configuration:
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 3]
```diff
@@ -932,6 +1088,45 @@ yum install httpd24-nghttp2
 systemctl restart httpd24-httpd
 ```
 
+</TabItem>
+
+<TabItem value="Debian 11" label="Debian 11">
+
+1. [Configure https on Centreon](#secure-the-web-server-with-https)
```
Comment (cg-tw) [inline versioned_docs/version-22.04/administration/secure-platform.md:1095]:
```suggestion
1. [Configure HTTPS on your Centreon server](#secure-the-web-server-with-https).
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 4]
```diff
@@ -932,6 +1088,45 @@ yum install httpd24-nghttp2
...
+
+2. Install nghttp2 module:
```
Comment (cg-tw) [inline versioned_docs/version-22.04/administration/secure-platform.md:1097]:
```suggestion
2. Install the nghttp2 module:
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 5]
```diff
@@ -932,6 +1088,45 @@ yum install httpd24-nghttp2
...
+
+```shell
+apt install nghttp2
+```
+
+3. Enable http2 protocol in **/etc/apache2/sites-available/centreon.conf**:
```
Comment (cg-tw) [inline versioned_docs/version-22.04/administration/secure-platform.md:1103]:
```suggestion
3. Enable the **http2** protocol in **/etc/apache2/sites-available/centreon.conf**:
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 6]
```diff
@@ -932,6 +1088,45 @@ yum install httpd24-nghttp2
...
+```apacheconf
+...
+<VirtualHost *:443>
+    Protocols h2 h2c http/1.1
+    ...
+</VirtualHost>
+...
+```
+
+4. Update method used by Apache multi-processus module in **/etc/apache2/sites-available/00-mpm.conf**:
```
Comment (cg-tw) [inline versioned_docs/version-22.04/administration/secure-platform.md:1114]:
```suggestion
4. Update the method used by the Apache multi-processus module in **/etc/apache2/sites-available/00-mpm.conf**:
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 7]
```diff
@@ -324,6 +351,24 @@ If you have SELinux installed, then update the SELinux policies:
 yum update -y selinux-policy*
 ```
 
+</TabItem>
+
+<TabItem value="Debian 11" label="Debian 11">
+
+```shell
+apt install epel-release
+apt install fail2ban fail2ban-systemd
```
Comment (lgcosta) [inline versioned_docs/version-22.04/administration/secure-platform.md:360]:
```suggestion
apt install fail2ban
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 8]
```diff
@@ -324,6 +351,24 @@ If you have SELinux installed, then update the SELinux policies:
...
+```
+
+If you have SELinux installed, then update the SELinux policies:
+
+```shell
+apt update -y selinux-policy*
+```
```
Comment (lgcosta) [inline versioned_docs/version-22.04/administration/secure-platform.md:367]:
```suggestion
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 9]
```diff
@@ -425,6 +470,14 @@ dnf install mod_ssl mod_security openssl
 yum install httpd24-mod_ssl httpd24-mod_security openssl
 ```
 
+</TabItem>
+
+<TabItem value="Debian 11" label="Debian 11">
+
+```shell
+apt install mod_ssl mod_security openssl
+```
```
Comment (lgcosta) [inline versioned_docs/version-22.04/administration/secure-platform.md:479]:
```suggestion
```shell
sudo systemctl stop apache2
sudo apt-get purge apache2 -y && sudo apt autoremove apache2 -y
curl -sSL https://packages.sury.org/apache2/README.txt | sudo bash -x
sudo apt update
sudo apt install apache2 -y
sudo apt install libapache2-mod-security2
sudo a2enmod security2
sudo systemctl restart apache2
```
```

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 10]
```diff
@@ -324,6 +351,24 @@ If you have SELinux installed, then update the SELinux policies:
...
+If you have SELinux installed, then update the SELinux policies:
+
+```shell
+apt update -y selinux-policy*
+```
+
+</TabItem>
+</Tabs>
+
 Enable fail2ban:
```
Comment (lgcosta) [inline versioned_docs/version-22.04/administration/secure-platform.md:372]: fail2ban ?

File: versioned_docs/version-22.04/administration/secure-platform.md [hunk 11]
```diff
@@ -532,6 +585,14 @@ dnf install mod_ssl mod_security openssl
 yum install httpd24-mod_ssl httpd24-mod_security openssl
 ```
 
+</TabItem>
+<TabItem value="Debian 11" label="Debian 11">
+
+```shell
+apt install mod_ssl mod_security openssl
+```
```
Comment (lgcosta) [inline versioned_docs/version-22.04/administration/secure-platform.md:593]:
```suggestion
```shell
sudo systemctl stop apache2
sudo apt-get purge apache2 -y && sudo apt autoremove apache2 -y
curl -sSL https://packages.sury.org/apache2/README.txt | sudo bash -x
sudo apt update
sudo apt install apache2 -y
sudo apt install libapache2-mod-security2
sudo a2enmod security2
sudo systemctl restart apache2
```
```

**Review by lgcosta — CHANGES_REQUESTED** [review]

**Review by lgcosta — DISMISSED** [review]

File: i18n/fr/docusaurus-plugin-content-docs/version-22.04/administration/secure-platform.md [hunk 12]
```diff
@@ -420,6 +462,21 @@ dnf install mod_ssl mod_security openssl
 yum install httpd24-mod_ssl httpd24-mod_security openssl
 ```
 
+</TabItem>
+
+<TabItem value="Debian 11" label="Debian 11">
+
+```shell
+sudo systemctl stop apache2
```
Comment (Dumbeldge) [inline i18n/fr/docusaurus-plugin-content-docs/version-22.04/administration/secure-platform.md:470]: After executing this block of command, i found an issue . We are supposed to modify the centreon.conf file later on in the documentation. After testing it twice, i found myself without the centreon.conf file at /etc/apache2/sites-available. Even when restarting with systemctl restart apache2, the file is not generated.

**Review by Dumbeldge — COMMENTED** [review]
After following the first part of the documentation, i found myself without the centreon.conf file that needs to be modified in order to properly implement https. I've commented the part that I suspect.
Also, for debian, we need to replace paths : 

-  "/etc/pki/tls/openssl.cnf" should be replaced by "/etc/ssl/openssl.cnf " for Debian
-  "/etc/pki/tls/private/" should be replaced by "/etc/ssl/private/" for Debian
- "/etc/pki/tls/certs/" should be replaced by "/etc/ssl/certs/" for Debian

File: i18n/fr/docusaurus-plugin-content-docs/version-22.04/administration/secure-platform.md (hunk 12 above)
Comment (lgcosta) [reply i18n/fr/docusaurus-plugin-content-docs/version-22.04/administration/secure-platform.md:470]: This file is generated in the installation of the `centreon-web-apache` package (verify installation) in the `/etc/apache2/sites-available`  directory, and then a symbolic link is created from it to `/etc/apache2/sites-enabled`:

````
root@centreon-lab-1:/etc/apache2/sites-enabled# ls -lha
total 8.0K
drwxr-xr-x 2 root root 4.0K Aug 30 00:08 .
drwxr-xr-x 8 root root 4.0K Aug 30 00:08 ..
lrwxrwxrwx 1 root root   32 Aug 12 15:58 centreon.conf -> ../sites-available/centreon.conf
```

**Review by lgcosta — CHANGES_REQUESTED** [review]
change `/etc/pki/tls/` all entries like Debian env (by comment of @Dumbeldge )

**Review by lgcosta — DISMISSED** [review]

**Review by tuntoja — DISMISSED** [review]

**Review by tuntoja — APPROVED** [review]

**Review by lgcosta — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**